pkg/registry/admissionreview/admission_review.go

```
@@ -0,0 +1,78 @@
+package admissionreview
+
+import (
+	"fmt"
+
+	"net/http"
+
+	"encoding/json"
+
+	admissionv1alpha1 "k8s.io/api/admission/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+type REST struct {
+}
+
+var _ rest.Creater = &REST{}
+
+func NewREST() *REST {
+	return &REST{}
+}
+
+func (r *REST) New() runtime.Object {
+	return &admissionv1alpha1.AdmissionReview{}
+}
+
+func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runtime.Object, error) {
+	fmt.Printf("#### got %#v\n", obj)
+
+	admissionReview := obj.(*admissionv1alpha1.AdmissionReview)
+	if admissionReview.Spec.Resource.Group != "project.openshift.io" ||
+		admissionReview.Spec.Resource.Resource != "projectrequests" ||
+		len(admissionReview.Spec.SubResource) != 0 ||
+		admissionReview.Spec.Operation != admissionv1alpha1.Create {
+
+		admissionReview.Status.Allowed = true
+		return admissionReview, nil
+	}
+
+	admittingObjectName := &NamedThing{}
+	err := json.Unmarshal(admissionReview.Spec.Object.Raw, admittingObjectName)
+	if err != nil {
+		return nil, errors.NewBadRequest(err.Error())
+	}
+
+	if len(admittingObjectName.Name) == 0 {
```

fail closed on empty name. might be generated later, but this fails anyway.